tests/testthat/test-literal_coercion_linter.R

```diff
@@ -29,22 +41,50 @@ patrick::with_parameters_test_that(
     literal_coercion_linter()
   ),
   .cases = tibble::tribble(
-        ~.test_name,   ~out_type,   ~input,
-    "lgl, from int",   "logical",     "1L",
-    "lgl, from num",   "logical",      "1",
-    "lgl, from chr",   "logical", '"true"',
-    "int, from num",   "integer",      "1",
-    "num, from num",   "numeric",      "1",
-    "dbl, from num",    "double",      "1",
-    "chr, from num", "character",      "1",
+    ~.test_name, ~out_type, ~input,
+    "lgl, from int", "logical", "1L",
+    "lgl, from num", "logical", "1",
+    "lgl, from chr", "logical", '"true"',
+    "int, from num", "integer", "1",
+    "num, from num", "numeric", "1",
+    "dbl, from num", "double", "1",
+    "chr, from num", "character", "1",
+    "chr, from chr", "character", '"e"',
+    "chr, from chr", "character", '"E"',
     # affirmatively lint as.<type>(NA) should be NA_<type>_
-    "int, from NA",    "integer",     "NA",
-    "num, from NA",    "numeric",     "NA",
-    "dbl, from NA",     "double",     "NA",
-    "chr, from NA",  "character",     "NA",
+    "int, from NA", "integer", "NA",
+    "num, from NA", "numeric", "NA",
+    "dbl, from NA", "double", "NA",
+    "chr, from NA", "character", "NA",
+  )
+)
+
+patrick::with_parameters_test_that(
+  "literal_coercion_linter blocks rlang disallowed usages",
+  expect_lint(
+    sprintf("%s(%s)", out_type, input),
+    rex::rex("Use literals directly where possible, instead of coercion."),
+    literal_coercion_linter()
+  ),
+  # even if `as.character(1)` works, `chr(1)` doesn't, so no corresponding test case
+  .cases = tibble::tribble(
+    ~.test_name,  ~out_type, ~input,
```

Comment:
can this be combined with the previous suite of parametrized tests?

Reply:
The key difference between these two tests is the `sprintf` function here:

- For base coercers

```r
  expect_lint(
    sprintf("as.%s(%s)", out_type, input),
    rex::rex("Use literals directly where possible, instead of coercion."),
    literal_coercion_linter()
  )
```

- For rlang coercers

```r
  expect_lint(
    sprintf("%s(%s)", out_type, input),
    rex::rex("Use literals directly where possible, instead of coercion."),
    literal_coercion_linter()
  )
```

So maybe a new column called `prefix` can be added with a value of `"as."` for base coercers, while `""` for rlang coercers, and then it will be `sprintf("%s%s(%s)", prefix, out_type, input),`.

WDYT?

Reply:
no that's overcomplicated, thanks, this works fine!